src/editor/FileSelector.vue

```diff
@@ -85,32 +100,35 @@ function horizontalScroll(e: WheelEvent) {
     @wheel="horizontalScroll"
     ref="fileSel"
   >
-    <div
-      v-for="(file, i) in files"
-      class="file"
-      :class="{ active: store.state.activeFile.filename === file }"
-      @click="store.setActive(file)"
-    >
-      <span class="label">{{
-        file === importMapFile ? 'Import Map' : file
-      }}</span>
-      <span v-if="i > 0" class="remove" @click.stop="store.deleteFile(file)">
-        <svg class="icon" width="12" height="12" viewBox="0 0 24 24">
-          <line stroke="#999" x1="18" y1="6" x2="6" y2="18"></line>
-          <line stroke="#999" x1="6" y1="6" x2="18" y2="18"></line>
-        </svg>
-      </span>
-    </div>
-    <div v-if="pending" class="file pending">
-      <input
-        v-model="pendingFilename"
-        spellcheck="false"
-        @blur="doneAddFile"
-        @keyup.enter="doneAddFile"
-        @keyup.esc="cancelAddFile"
-        @vnodeMounted="focus"
-      />
-    </div>
+    <template v-for="(file, i) in files">
+      <div
+        v-if="pending !== file"
+        class="file"
+        :class="{ active: store.state.activeFile.filename === file }"
+        @click="store.setActive(file)"
+        @dblclick="i > 0 && editFileName(file)"
```

Comment:
I can see why that might be desirable, but what I've done here is consistent with deleting a file.

My reasoning is that it's already possible to rename a file by deleting it and re-adding it. It's painful for the user, but it is possible. The only restriction we currently place on that in the UI is preventing the first file from being deleted. So I've maintained the same restriction for renaming.

I think there is potential for providing various forms of permissions, to restrict what files can be added/removed/renamed, but currently we don't have anything like that and I can't think of a reason why it would be necessary to add it specifically for renaming when it isn't available for deleting.